source/common/network/utility.cc

```diff
@@ -66,13 +66,30 @@ std::string hostFromUrl(const std::string& url, absl::string_view scheme,
     throw EnvoyException(fmt::format("expected {} scheme, got: {}", scheme_name, url));
   }
 
-  const size_t colon_index = url.find(':', scheme.size());
+  const size_t bracket_begin_index = url.find('[', scheme.size());
 
-  if (colon_index == std::string::npos) {
-    throw EnvoyException(absl::StrCat("malformed url: ", url));
-  }
+  if (bracket_begin_index == std::string::npos) {
+    const size_t colon_index = url.find(':', scheme.size());
+
+    if ((colon_index == std::string::npos) || (colon_index == scheme.size())) {
```

Comment:
Why does colon_index == scheme.size() indicate a malformed URL? Comment would be good